appendices/VK_EXT_custom_border_color.txt

```diff
@@ -0,0 +1,89 @@
+include::{generated}/meta/VK_EXT_custom_border_color.txt[]
+
+*Last Modified Date*::
+    2019-10-14
+*IP Status*::
+    No known IP claims.
+*Contributors*::
+  - Joshua Ashton
+  - Liam Middlebrook, Nvidia
+
+This extension provides cross-vendor functionality to specify a custom border color
+for use when the sampler address mode ename:VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_BORDER is used.
+
+To create a sampler which uses a custom border color
+slink:VkSamplerCreateInfo::pname:borderColor must be set to ename:VK_BORDER_COLOR_CUSTOM_EXT.
+
+When ename:VK_BORDER_COLOR_CUSTOM_EXT is supplied, applications must provide a
+slink:VkSamplerCustomBorderColorCreateInfoEXT in the pNext chain for
+slink:VkSamplerCreateInfo.
+
+For implementations with a limited number of border colors, they should be reference counted
+when the sampler is created/destroyed. This means that freeing the last sampler that used
+a custom border color will also free up the slot for another border color to be used.
+
+The implementation should maintain the guarantee that flink:vkCreateSampler and
+flink:vkDestroySampler is thread-safe when a custom border color is used.
+
+=== New Enum Constants
+
+  * Extending elink:VkStructureType:
+  ** ename:VK_STRUCTURE_TYPE_SAMPLER_CUSTOM_BORDER_COLOR_CREATE_INFO_EXT
+  ** ename:VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_CUSTOM_BORDER_COLOR_PROPERTIES_EXT
+  ** ename:VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_CUSTOM_BORDER_COLOR_FEATURES_EXT
+
+  * Extending elink:VkBorderColor:
+  ** ename:VK_BORDER_COLOR_CUSTOM_EXT
+
+=== New Structures
+
+  * slink:VkSamplerCustomBorderColorCreateInfoEXT
+  * slink:VkPhysicalDeviceCustomBorderColorPropertiesEXT
+  * slink:VkPhysicalDeviceCustomBorderColorFeaturesEXT
+
+=== New Functions
+
+=== Issues
+
+1) Should VkClearColorValue be used for the border color value, or should we
+have our own struct/union?
+Do we need to specify the type of the input values for the components? This is
+more of a concern if VkClearColorValue is used here because it provides a union
+of float,int,uint types.
```

Comment:
I think the right answer is probably to have three new VK_BORDER_COLOR_CUSTOM enums to match the three types, because otherwise the driver wouldn't know which of the three union members to read from (we've been careful to avoid relying on type punning via unions).

Reply:
Yeah, that makes sense. Should also help with a possible Intel implementation if I understand their HW docs correctly.

Will sort that 😊